npm/webpack-preprocessor/lib/cross-origin-callback-store.ts

```diff
@@ -0,0 +1,29 @@
+export interface CrossOriginCallbackStoreFile {
+  inputFileName: string
+  outputFilePath: string
+  source: string
+}
+
+export class CrossOriginCallbackStore {
+  private files: { [key: string]: CrossOriginCallbackStoreFile[] } = {}
+
+  addFile (sourceFilePath: string, file: CrossOriginCallbackStoreFile) {
+    this.files[sourceFilePath] = (this.files[sourceFilePath] || []).concat(file)
+  }
+
+  hasFilesFor (sourceFiles: string[]) {
+    return !!this.getFilesFor(sourceFiles)?.length
+  }
+
+  getFilesFor (sourceFiles: string[]) {
+    return Object.keys(this.files).reduce((files, sourceFilePath) => {
+      return sourceFiles.includes(sourceFilePath) ? files.concat(this.files[sourceFilePath]) : files
+    }, [] as CrossOriginCallbackStoreFile[])
+  }
+
+  reset (sourceFilePath: string) {
+    this.files[sourceFilePath] = []
```

Comment:
Should this also delete the files from the file system or when will those be cleaned up?

Reply:
Realized we have a perfectly good place to clean them up: when the preprocessor receives the 'close' event. Addressed in [`95e7c65` (#23283)](https://github.com/cypress-io/cypress/pull/23283/commits/95e7c65080fc66040bdea8f1c8d8ba5e101c8ef7)